Utilities/bootstrap

```diff
@@ -1052,6 +1055,9 @@ def main():
 
     env_cmd = ["env", "SWIFT_EXEC=" + os.path.join(bindir, "swiftc"),
                       "SWIFTPM_BUILD_DIR=" + build_path]
+
+    if args.swift_lib_dir:
+        env_cmd.append("LD_LIBRARY_PATH=" + args.swift_lib_dir)
```

Comment:
Can we put a symlink in the fake toolchain instead of using LD_LIBRARY_PATH or that doesn't work?

Reply:
We have several tests in SwiftPM that build Swift packages in debug mode. I don't understand why we don't hit the same problem there. @jrose-apple said these libraries can be considered part of stdlib.

Reply:
I don't think it matters whether the packages that SwiftPM builds are in debug or release mode but whether the `libFoundation.so` they link to is in release or debug mode. In debug mode it links to `libswiftSwiftOnone.so` which it can't find in the build directories without this path.
The `--debug-foundation` flag controls whether the `libFoundation.so` is built in debug mode.

Reply:
@jrose-apple said these libraries are required if something is being built with `-Onone` and we have several test packages that are built with `-Onone`. Presence of libFoundation.so shouldn't matter.

Reply:
Can we have an investigation on this behavior and why and how it affects SwiftPM tests? It is blocking Foundation adoption of `@testable import`.

Reply:
When you build with -Onone, you gain a dependency on libswiftSwiftOnoneSupport.so/.a (yes, it's silly that there are two "swift"s in there). It shouldn't have anything to do with `@testable import`.

Reply:
I'll take a closer look tomm...